src/main/scala/com/lambtors/poker_api/module/shared/domain/DeckProvider.scala

```diff
@@ -0,0 +1,8 @@
+package com.lambtors.poker_api.module.shared.domain
+
+import com.lambtors.poker_api.module.poker.domain.model.Card
+
+trait DeckProvider {
+  def provide(): List[Card]
+  def shuffleGivenDeck(deck: List[Card]): List[Card]
```

Comment:
remove this method, if we need to shuffle a deck, maybe have a DeckShuffler or something, DeckProvider should be focused on providing decks